src/python/roms/__init__.py

```diff
@@ -1,141 +1,90 @@
-import functools
-import logging
-import os
-import pathlib
-import sys
+from __future__ import annotations
+
+import base64
+import hashlib
+import json
+import tarfile
 import warnings
-from typing import List, Optional
-
-from ale_py import ALEInterface
-from ale_py.roms import plugins, utils
-
-# pylint: disable=g-import-not-at-top
-if sys.version_info >= (3, 9):
-    import importlib.resources as resources
-else:
-    import importlib_resources as resources
-# pylint: enable=g-import-not-at-top
-
-# Precedence is as follows:
-#  1. Internal ROMs
-#  2. External ROMs
-#  3. ROMs from atari-py.roms
-#  4. ROMs from atari-py-roms.roms
-_ROM_PLUGIN_REGISTRY: List[plugins.Plugin] = [
-    plugins.Package("ale_py.roms"),
-    plugins.EntryPoint("ale_py.roms"),
-    plugins.Package("atari_py.atari_roms"),
-    plugins.Package("atari_py_roms.atari_roms"),
-]
-
-# Environment variable for ROM discovery.
-# ale-py will search for supported ROMs in:
-#   ${ALE_PY_ROM_DIR}/*.bin
-_ROM_DIRECTORY_ENV_KEY = "ALE_PY_ROM_DIR"
-_ROM_DIRECTORY_ENV_VALUE = os.environ.get(_ROM_DIRECTORY_ENV_KEY, None)
-
-if _ROM_DIRECTORY_ENV_VALUE is not None:
-    _ROM_PLUGIN_REGISTRY.append(plugins.Directory(_ROM_DIRECTORY_ENV_VALUE))
-
-
-def _resolve_rom(name: str) -> Optional[pathlib.Path]:
-    """Resolve a ROM path from the ROM registry."""
-    for package in _ROM_PLUGIN_REGISTRY:
-        rom_id = utils.rom_name_to_id(name)
-
-        # Resolve ROM from package
-        try:
-            rom_path = package.resolve(rom_id)
-        except ModuleNotFoundError:
-            continue
+from pathlib import Path
 
-        # Failed to resolve ROM
-        if rom_path is None:
-            logging.debug(f"{package} did not resolve {rom_id}.")
-            continue
+import requests
 
-        # ROM isn't supported
-        resolved_id = ALEInterface.isSupportedROM(rom_path)
-        if resolved_id is None:
-            warnings.warn(
-                f"{package} contains unsupported ROM: {rom_path}",
-                category=ImportWarning,
-                stacklevel=2,
-            )
-            continue
 
-        # If the ROMs resolved differently
-        if resolved_id != rom_id:
-            warnings.warn(
-                f"{package} contains ROM {rom_id} which doesn't resolve to {resolved_id}. "
-                "This is most likely caused by a filename mismatch.",
-            )
+def _download_roms():
+    """Downloads all roms in the form of a base64 file, then decodes it into a tar.gz, then extracts all the roms by matching it to the expected md5."""
+    # this is a map of {rom.bin : md5 checksum}
+    all_roms = json.load(open(Path(__file__).parent / "md5.json"))
```

Comment:
Use `with open()` such that the file is closed correctly

Reply:
Don't think this is needed, since we pass the fp directly into json and the `load` function closes after reading the files, garbage collector should automatically close it for us.

Reply:
Just to make sense it is correct, can you change